Dockerfile

```diff
@@ -0,0 +1,34 @@
+FROM python:buster as base
+
+# For Production
+
+FROM base as production
+
+RUN apt-get update && apt-get install -y curl
+RUN curl -sSL https://install.python-poetry.org | /usr/bin/python3 -
+RUN mkdir -p /app/todo_app
+COPY poetry* /app
+COPY pyproject.toml /app
+COPY todo_app /app/todo_app
+WORKDIR /app
+RUN ~/.local/bin/poetry install
+ENV poetry=/root/.local/bin
+ENV PATH=${poetry}:${PATH}
+CMD poetry run gunicorn --bind 0.0.0.0:5000 "todo_app.app:create_app()"
+EXPOSE 5000/tcp
+
+# For Development
+
+FROM base as development
```

Comment:
Although this works, it's missing a bit part of the advantage of having a multi-stage Dockerfile; reducing duplication!

The first few instructions here are the same, so making those shared would make sense, e.g. having at the top:

```Dockerfile
FROM python:buster as base

RUN apt-get update && apt-get install -y curl
RUN curl -sSL https://install.python-poetry.org | /usr/bin/python3 -
# For Production

FROM base as production

RUN mkdir -p /app/todo_app
...
```
Would let us skip the first couple of instructions here. Noting that Docker also lets us remove the `mkdir` command if we're careful (see above) would let us share a few more instructions too